### PR TITLE
Buncha small stuff; most importantly single file download by snap_id

### PR DESCRIPTION
**Pull request by rozbb**
Pretty self-explanatory

One main point I'd like to defend is the use of `__future__` in `snapchatfs.py`. By consolidating the session management I ran into a problem on [this line](https://github.com/doomrobo/snapchat-fs/compare/hausdorff:master...master#diff-b81d0597a19ae50ecd0574f0ee6b285eR114) whereby isinstance wasn't working. After doing some digging I came upon an [SO thread](http://stackoverflow.com/questions/13039060/proper-use-of-isinstanceobj-class) that said the problem was due to double-importing which itself was a result of importing snapchat_core twice (once in `sfs` and once in `snapchatfs.py`, or once in `snapchat_fs/__init__.py` or...I don't know). By using absolute imports the problem is fixed.

The bottom line on the README changed because Vim ended the file (as-per the POSIX standard) with a newline.

Also I changed all the files to mode 755 because they're not editable for anyone but root as 644 (the previous mode).

### CODE REVIEW

**Comment by rozbb** [comment]
Just finished merging with the repo. sfs doesn't appear to have any changes, though. It was mostly `snapchatfs.py`

**Comment by hausdorff** [comment]
- Ok, reviewing this now.
- Hahaha, ok well in my defense it was literally 3AM when I read it, and I was not really together on it.

**Comment by hausdorff** [comment]
Thanks again for this patch. It's really good work.

**Comment by rozbb** [comment]
No problem! As soon as I can get single-file deletion working, I'll PR that too (it's not working for some reason; 400 error on POST)
